test/integration/framework/etcd.go

```diff
@@ -129,8 +129,17 @@ func RunCustomEtcd(dataDir string, customFlags []string) (url string, stopFn fun
 	cmd.Stderr = os.Stderr
 	stop := func() {
 		cancel()
-		err := cmd.Wait()
-		klog.Infof("etcd exit status: %v", err)
+		errCh := make(chan error, 1)
+		go func() {
+			errCh <- cmd.Wait()
+		}()
+		select {
+		case err := <-errCh:
+			klog.Infof("etcd exit status: %v", err)
+		case <-time.After(60 * time.Second):
+			err := cmd.Process.Kill()
```

Comment:
hmm the context kills the process too , but at least this wait we are not blocked by wait